src/osp/Resource/blueprints.cpp

```diff
@@ -56,12 +56,20 @@ void BlueprintVehicle::add_part(
 
     // now we have a part index.
 
-    // just create a new part blueprint will all the data
+    // Create and default initialize object blueprint machines
+    size_t numMachines = prototype->get_machines().size();
+    std::vector<BlueprintMachine> machineBPs(numMachines, BlueprintMachine{});
```

Comment:
I believe that std::vector::resize would accomplish this goal slightly more efficently.

Reply:
Resize where? You mean like
```cpp
std::vector<BlueprintMacine> machineBPs;
machineBPs.resize(numMacines);
```
?

Reply:
Yep.

Resize calls default constructor directly.

Your code is using the copy constructor from a default value.

A smart optimizer should figure it out, but not all optimizers are smart, so better to make it do fewer things. Less places to get wrong.

Then again, maybe it makes zero difference. Its always hard to really predict these things.